src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    const newState = { ...currentState };
+
+    if (action.type === 'clear') {
+      for (const key in newState) {
+        delete newState[key];
+      }
+    } else if (action.type === 'addProperties') {
+      Object.assign(newState, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((key) => {
+        if (newState.hasOwnProperty(key)) {
+          delete newState[key];
+        }
+      });
+    }
```

Comment:
looks like it should work, but for better developer experience I'd rather use switch/case instead